app/src/main/java/com/example/weather/WeatherRequest.java

```diff
@@ -28,6 +30,46 @@
         this.location = location;
     }
 
+    private WeatherRequest(Parcel parcel) {
+        readFromParcel(parcel);
+    }
+
+    private void readFromParcel(Parcel parcel) {
+        this.weatherApiKey = parcel.readString();
+        this.weatherApiEntryPoint = parcel.readString();
+        this.cityName = parcel.readString();
+        this.location = Location.CREATOR.createFromParcel(parcel);
+    }
+
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel parcel, int flags) {
+        parcel.writeString(weatherApiKey);
+        parcel.writeString(weatherApiEntryPoint);
+        parcel.writeString(cityName);
+        parcel.writeParcelable(location, flags);
```

Comment:
Please, help me deal with issue in this place.
Parcel object Location in this way cause null object reference in MainActivity with IWeatherServiceCallback.
If parcel object Location as commented in lines 55-58 - without any error.

Reply:
thanks